src/core/evaluator.js

```diff
@@ -402,6 +403,25 @@ class PartialEvaluator {
     return data;
   }
 
+  loadFontInDocument(name) {
+    const cachedData = this.standardFontDataCache.get(name);
+    if (cachedData) {
+      return cachedData;
+    }
```

Comment:
Sorry, but I really don't see how it's generally correct/safe to re-use that cache here!

A new `this.fetchStandardFontData`-call, from *another* PDF document font, may now end up returning *completely unexpected* data.
Please introduce a new cache for this use-case, following the same patterns as this cache, to avoid any issues now or later; I'd suggest calling it `FontSubstitutionCache`.